src/server/table.h

```diff
@@ -76,7 +76,77 @@ struct DbTableStats {
   DbTableStats& operator+=(const DbTableStats& o);
 };
 
-using LockTable = absl::flat_hash_map<std::string, IntentLock>;
+// We use LockKey for LockTable keys because of the multi transactions
+// that unlock asynchronously. We must ensure the existence of keys outside of
+// multi-transaction lifecycle.
+
```

Comment:
We already do! We have `flat_hash_set<string> MultiData::lock_keys` and it would be easier to "launder" lock keys instead of using a variant here

Reply:
I thought about it too! Did not find an easy way to actually do it. It's all yours.